torch/distributed/_functional_collectives.py

```diff
@@ -546,3 +551,25 @@ def _register_ops():
     _register_ops()
 else:
     warnings.warn("PyTorch Distributed functional collectives do not work with torch::deploy.")
+
+
+def _all_gather_into_tensor_coalesced_fallback(output_tensors, input_tensors, group, async_op=False):
+    # all_gather_coalesced is useless, it doesn't work under NCCL and does lots of copies under Gloo
+    # all_gather is useless too because it's single tensor
+    # NCCL's PG::all_gather with multiple tensors is broken, it only works for the multi-device setting
+    #  and fails if you mix same-size with different-size tensor lists.
+    # _coalescing_manager crashed NCCL when used with all_gather_into_tensor.
```

Comment:
I guess we should use coalescing manager once this issue is fixed?

Reply:
Yes, hopefully we'll use coalescing manager or something similar once it's available